### PR TITLE
added all filters

### PR DESCRIPTION
**Pull request by tushargoyal812**
https://user-images.githubusercontent.com/69681896/159791032-78f9cce6-00c2-42bc-9e2c-f7955e55cda2.mp4

### CODE REVIEW

File: src/filter-functions/androidHandler.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+//     {
+//         return data
+//     }
+//     if(state.category.android)
+//     {
+//         arr=[...data].filter(item=>item.android)
+//         return arr
+//     }else{
+//         return arr
+//     }
```
Comment (sharmayatendra) [inline src/filter-functions/androidHandler.js]: you can remove commented code

File: src/filter-functions/category.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    let arr=[]
+    if(!state.category.ios&&!state.category.android)
+    {
+        return data
+    }
+    if(state.category.ios)
+    {
+        let newArr=[...data].filter(item=>item.ios)
+        arr.push(...newArr)
+        console.log('ios wala',arr);
```
Comment (sharmayatendra) [inline src/filter-functions/category.js]: remove all console.log

File: src/filter-functions/price-range.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        return data
+    }
+    }
+    //    switch (state.priceRange) {
+    //        case 10:  
+    //           return [...data].filter(item=>item.price>10000)
+
+    //        default:
+    //            return data;
+    //    }
```
Comment (sharmayatendra) [inline src/filter-functions/price-range.js]: same as above remove commented code

File: src/pages/products/products.js [hunk 4]
```diff
@@ -5,11 +5,24 @@ import { Sidebar } from '../../components/sidebar/sidebar'
 import { useFilter } from '../../filter-context/filter-context'
 import { highToLow } from '../../filter-functions/high-to-low'
 import {lowToHigh} from '../../filter-functions/low-to-high'
+import { inStock } from '../../filter-functions/in-stock'
+import { fastDelivery } from '../../filter-functions/fast-delivery'
+import { priceRange } from '../../filter-functions/price-range'
+import { rating } from '../../filter-functions/rating'
+import { categoryHandler } from '../../filter-functions/category'
```
Comment (sharmayatendra) [inline src/pages/products/products.js]: you can use all these by putting them into a single file & then importing them

**Review by sharmayatendra — APPROVED** [review]
1. code looks clean everything is understood like variable names are good.
2. file structure is also well maintained.
3.  all the functions are isolated separately that's good

File: src/filter-functions/price-range.js (hunk 3 above)
Comment (tushargoyal812) [reply src/filter-functions/price-range.js]: done

File: src/pages/products/products.js (hunk 4 above)
Comment (tushargoyal812) [reply src/pages/products/products.js]: ok got it.

File: src/filter-functions/category.js (hunk 2 above)
Comment (tushargoyal812) [reply src/filter-functions/category.js]: ok

File: src/filter-context/filter-context.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ import { reducerFunc } from '../reducer-func/reducer-func'
 const FilterContext=createContext()
 
 const FilterProvider=({children})=>{
-    const [state, dispatch] = useReducer(reducerFunc, {sorting:null})
+    const [state, dispatch] = useReducer(reducerFunc, {sorting:null,isInStock:false,fastDelivery:false,priceRange:50000,rating:5,category:{ios:false,android:false}})
```
Comment (arjunbharti) [inline src/filter-context/filter-context.js:7]: you can create an object called `initialState` and extract out these properties to make the code more clean. Although, it is fine too.

File: src/filter-functions/androidHandler.js (hunk 1 above)
Comment (arjunbharti) [inline src/filter-functions/androidHandler.js]: if this code is not required then better to delete this file.

File: src/filter-functions/category.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+export const categoryHandler=(data,state)=>{
+    let arr=[]
```
Comment (arjunbharti) [inline src/filter-functions/category.js]: try to be more specific with naming of objects and arrays.
For example - you can write it as `categoryList`

File: src/filter-functions/price-range.js (hunk 3 above)
Comment (arjunbharti) [inline src/filter-functions/price-range.js]: avoid pushing unwanted comments

**Review by arjunbharti — APPROVED** [review]
Logic looks right to me. Nice work
Few suggestions:

- Be more specific when naming any object and array in the project
- avoid pushing unwanted comments
- Import named exports at one place and import a single file as mentioned by another reviewer also

File: src/filter-context/filter-context.js (hunk 5 above)
Comment (tushargoyal812) [reply src/filter-context/filter-context.js:7]: ok will do it

File: src/filter-functions/category.js (hunk 6 above)
Comment (tushargoyal812) [reply src/filter-functions/category.js]: right will do it

File: src/filter-functions/price-range.js (hunk 3 above)
Comment (tushargoyal812) [reply src/filter-functions/price-range.js]: removed it

File: src/filter-functions/androidHandler.js (hunk 1 above)
Comment (tushargoyal812) [reply src/filter-functions/androidHandler.js]: yes got it

**Pull request merged**